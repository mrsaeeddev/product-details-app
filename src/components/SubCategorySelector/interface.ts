export interface subCategories {
    subcategories: string[];
    handlePress(v: string): void;
}