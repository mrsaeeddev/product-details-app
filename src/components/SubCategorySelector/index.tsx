import React from "react";
import { View, StyleSheet } from "react-native";
import SubCategory from "../SubCategory";
import { globalStyles } from "../../globalStyles";
import { subCategories } from "./interface";

const styles = StyleSheet.create({
    grid: {
        flex: 1,
        flexDirection: 'row',
        flexWrap: 'wrap',
    },
})

function SubCategorySelector(props: subCategories) {
    return (<View style={[globalStyles.container, styles.grid]}>{props.subcategories.map((v, i) => <SubCategory subcategory={v} />)}</View>)
}

export default SubCategorySelector;
