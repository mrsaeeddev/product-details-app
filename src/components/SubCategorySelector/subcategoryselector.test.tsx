import React from 'react';
import renderer from 'react-test-renderer';

import SubCategorySelector from './';

describe('SubCategory', () => {
    it('should render correctly using Snapshot', () => {
        const subcategories = ['dummy1', 'dummy2']
        const handlePress = jest.fn()
        const handleAllProducts = jest.fn()
        const tree = renderer.create(<SubCategorySelector subcategories={subcategories} handleAllProducts={handleAllProducts} handlePress={handlePress} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly using Snapshot', () => {
        const subcategories = ['dummy1', 'dummy2']
        const handlePress = jest.fn()
        const handleAllProducts = jest.fn()
        const tree = renderer.create(<SubCategorySelector subcategories={subcategories} handleAllProducts={handleAllProducts} handlePress={handlePress} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly using Snapshot', () => {
        const subcategories = ['dummy1', 'dummy2']
        const handlePress = jest.fn()
        const handleAllProducts = jest.fn()
        const tree = renderer.create(<SubCategorySelector subcategories={subcategories} handleAllProducts={handleAllProducts} handlePress={handlePress} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});