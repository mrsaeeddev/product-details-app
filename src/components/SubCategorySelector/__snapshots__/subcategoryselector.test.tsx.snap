// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategory should render correctly using Snapshot 1`] = `
<View>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginLeft": 15,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "backgroundColor": "#F5F5F5",
                  "borderRadius": 10,
                  "height": 50,
                  "marginRight": 15,
                  "resizeMode": "stretch",
                  "width": 225,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              All Products
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`SubCategory should render correctly using Snapshot 2`] = `
<View>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginLeft": 15,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "backgroundColor": "#F5F5F5",
                  "borderRadius": 10,
                  "height": 50,
                  "marginRight": 15,
                  "resizeMode": "stretch",
                  "width": 225,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              All Products
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`SubCategory should render correctly using Snapshot 3`] = `
<View>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginLeft": 15,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "backgroundColor": "#F5F5F5",
                  "borderRadius": 10,
                  "height": 50,
                  "marginRight": 15,
                  "resizeMode": "stretch",
                  "width": 225,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              All Products
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 10,
                "height": 50,
                "marginRight": 15,
                "resizeMode": "stretch",
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            dummy2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
